backend/models/node.py

```diff
@@ -1,87 +1,249 @@
-from .shared import db
-from .measurement import Measurement
+from .shared import db # Allows the models to be split out into separate files.
 
 
 class Node(db.Model):
-    __tablename__ = "node"
+    """
+    A class used to generate a table of Nodes.
+    Nodes are used to model Hierarchy trees that consist of objectives,
+    sub-objectives, and terminating in measurements.
+
+    Based off of SQLALchemy's adjacency_list.py:
+    https://docs.sqlalchemy.org/en/14/_modules/examples/adjacency_list/adjacency_list.html
+
+    ...
+
+    Attributes
+    ----------
+    id : int
+        Node's unique id.
+    parent_id : int
+        Parent node's unique id.
+    hierarchy_id : int
+        Id of the hierarchy the node belongs to.
+    name : str
+        The name of the node.
+    icon : str
+        String to store the icon the frontend uses for the node.
+    weight : float
+        Floating point value representing weight
+    is_measurement : bool
```

Comment:
I'm starting to think that this flag is a bad idea (my fault) because there's a possibility that is_measurement and measurement_type aren't in sync. So the front end could either just check measurement_type is null, or it would check is_measurement and have trust the measurement_type is not null.